src/fileformats/ocd_file_import.cpp

```diff
@@ -2449,4 +2412,57 @@ void OcdFileImport::handleStrings(const OcdFile<F>& file, std::initializer_list<
 }
 
 
+
+OcdFileImport::OcdParameterStreamReader::OcdParameterStreamReader(const QString& param_string) : param_string(param_string)
+{
+	param_ptr = 0;
+}
+
+bool OcdFileImport::OcdParameterStreamReader::readNext()
+{
+	while (!atEnd())
+	{
+		param_ptr = param_string.indexOf(QLatin1Char('\t'), param_ptr);
+		if (param_ptr < 0 || param_ptr + 1 >= param_string.length())	// accept \t at end only if there is space for at least the key
+		{
+			param_ptr = param_string.length();
+			return false;
+		}
+		++param_ptr;
+		if (param_string.at(param_ptr).toLatin1() != '\t')	// is the next key \t ? if yes then skip over
+			return true;
+	}
+	return false;
+}
+
+char OcdFileImport::OcdParameterStreamReader::key() const
+{
+	if (atEnd())
+		return '\0';	// function needs to return something
+	if (param_ptr)
+		return param_string.at(param_ptr).toLatin1();
+	else
+		return '\f';	// return generic '\f' for 'first' parameter
```

Comment:
The idea behind the \f is that processing of the 'first' parameter could also be done inside the loop:
```
const QString name = parameters.value().toString();		// 'first' parameter
while (parameters.readNext())
{
switch (parameters.key())
		{
		case 'n':
			number = param_value.toInt(&number_ok);
			break;
```
could be changed to
```
do
{
switch (parameters.key())
		{
		case '\f':
			name = parameters.value().toString();	// 'first' parameter
			break;
		case 'n':
			number = param_value.toInt(&number_ok);
			break;
.....
} while (parameters.readNext());
```
Since \f is not used as a regular key (but f is) I decided to chose \f in allusion to 'first'.
If the parameter string would be empty, the generic '\0' would fall through and the while below end any futher parsing.

Reply:
Okay, `'\f'  != 'f'`.

If we want a cute name, we should use a `constexpr` member function returning the desired character. And `\0` would still work. Reading at the end is an invalid operation anyways, so there is no conflict.

Reply:
I think the implementation of '\f' is acceptable since the scope of that helpe class is very limited and since '\f' is actually not used by the parent class when processing parameter strings. Besides that, the arbitrarily chosen key replacement is well documented.
I don't know how to implement such a constexpr member function, so please make a proposal.